tools/chromium_src/check_chromium_src.py

```diff
@@ -0,0 +1,543 @@
+                    if not self.validate_exclusion_symbol(path, symbol):
+                        result = False
+
+        return result
+
+    def load_exclusions(self):
+        config_path = os.path.join(BRAVE_SRC, 'check_chromium_src_config.json5')
+        if not os.path.isfile(config_path):
+            self.AddError(f"  Unable to load config file {config_path}.")
+            return False
+
+        try:
+            json5_path = os.path.join(CHROMIUM_SRC, 'third_party', 'pyjson5',
+                                      'src')
+            sys.path.append(json5_path)
+            # pylint: disable=import-outside-toplevel,import-error
+            import json5
+
+            with open(config_path, "r", encoding='utf-8') as exclusions_file:
+                self.config_data = json5.load(exclusions_file)
+        finally:
+            # Restore sys.path to what it was before.
+            sys.path.remove(json5_path)
+
+        keys = ['re_excludes', 'path_excludes', 'symbol_excludes']
+        for key in keys:
+            if not key in self.config_data:
+                self.AddError(f"  Key '{key}' is missing from {config_path}.")
+                return False
+
+        return self.validate_exclusions()
+
+    def check_overrides(self, affected_paths=None):
+        # Load and check exclusions
+        if not self.load_exclusions():
+            return self.messages
+
+        # Change into the chromium_src directory for convenience.
+        os.chdir(BRAVE_CHROMIUM_SRC)
+
+        # Build filters
+        exclude_regexp = None
+        if (len(self.config_data['re_excludes']) > 0
+                or len(self.config_data['path_excludes']) > 0):
+            exclude_regexp = '|'.join(self.config_data['re_excludes'] +
+                                      self.config_data['path_excludes'])
+
+        # Build the list of files to check.
+        if affected_paths is None:
+            self.overrides = filter_all_chromium_src_filepaths(exclude_regexp)
+        else:
+            self.overrides = filter_chromium_src_filepaths(
+                affected_paths, exclude_regexp)
+
+        # Check overrides
+        if len(self.overrides) > 0:
+            self.do_check_overrides()
+
+        return self.messages
+
+
+def main(args):
+    """
+    Parse command line args and check all overrides in chromium_src.
+    """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('build',
+                        help='Build type (i.e. Component|Static|Debug|Release)',
+                        nargs='?',
+                        default='Component')
+    parser.add_argument('--os',
+                        required=False,
+                        help='Target OS (e.g. android|win|linux|mac|ios)')
+    parser.add_argument('--arch',
+                        required=False,
+                        help='Target architecture (e.g. x86|x64|arm|arm64)')
+    options = parser.parse_args(args)
```

Comment:
do we really need to pass `args` here? it should work without this.
https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.parse_args

Reply:
Yep, not needed. Fixed in dde637633ee8265bb9dbc7a7b39cc41c4c46d9a1